Mlem/Views/Tabs/Feeds/Feed View.swift

```diff
@@ -39,30 +41,47 @@ struct FeedView: View {
     init(
         community: CommunityModel?,
         feedType: FeedType,
-        sortType: PostSortType,
         showLoading: Bool = false
     ) {
         // need to grab some stuff from app storage to initialize post tracker with
         @AppStorage("internetSpeed") var internetSpeed: InternetSpeed = .fast
         @AppStorage("upvoteOnSave") var upvoteOnSave = false
         
+        self.showLoading = showLoading
+        
         self._feedType = State(initialValue: feedType)
-        self._postSortType = .init(initialValue: sortType)
         self._postTracker = StateObject(wrappedValue: .init(
             shouldPerformMergeSorting: false,
             internetSpeed: internetSpeed,
             upvoteOnSave: upvoteOnSave
         ))
         
-        self.showLoading = showLoading
         self._community = State(initialValue: community)
+        
+        @Dependency(\.siteInformation) var siteInformation
+        @AppStorage("defaultPostSorting") var defaultPostSorting: PostSortType = .hot
+        @AppStorage("fallbackDefaultPostSorting") var fallbackDefaultPostSorting: PostSortType = .hot
+        
+        if let siteVersion = siteInformation.version {
+            
+            if siteVersion >= defaultPostSorting.minimumVersion {
+                _postSortType = .init(initialValue: defaultPostSorting)
+            } else {
+                _postSortType = .init(initialValue: fallbackDefaultPostSorting)
+            }
+            isWaitingForSiteInformation = false
+        } else {
+            _postSortType = .init(initialValue: .hot)
+            isWaitingForSiteInformation = true
+        }
```

Comment:
```suggestion
        _postSortType = .init(initialValue: fallbackDefaultPostSorting)
        isWaitingForSiteInformation = true
```

Occasionally the feed ends up set to `.hot` and never updates thereafter. I think this is caused by siteInformation updating after the check here but before the view renders, leading to it not getting picked up by the .onChange handler. I've developed a fix that seems to work--see below for the other half of it